packages/opentelemetry-types/src/distributed_context/distributed_context.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Entry, EntryMetadata} from './entry';
+
+export interface DistributedContext {
```

Comment:
This is along the lines of @rochdev suggestions for simplifying the interface: what would you think about making this an object where the key of the object is `key` and the value is something like `EntryValue`, that has a `value` field and a `ttl` field? So something like this:

```
export type DistributedContext {
   [key: string]: EntryValue
}

export interface EntryValue {
  value: string;
  ttl?: EntryTtl;
}
```

Would that be too much of a deviation from the spec? To me, the simpler object feels a little more TS/JS friendly than the extra layers of objects and the `put`/`remove`/`getEntries` methods, which looks more like an idiomatic Java approach, IMO.

It would also allow the easy creation of a distributed context just by setting up an object, so `{key1: {value: 'abc', ttl:  EntryTtl. NO_PROPAGATION }}` can be a valid context without needing to call any builder functions to set it up.

Reply:
It could even be simplified to:

```ts
export type DistributedContext {
   [key: string]: EntryValue | string
}
```

Or even `unknown` instead of `string` if possible. Then when you don't need the `ttl`, which should be almost always (for Datadog it will be 100% of the time that we wouldn't need that as I still don't understand why that's even a thing) then you have a much simpler interface to work with.

Reply:
I like the idea of being able to specify a `string` to `string` mapping for convenience, but then that also adds complexity in how this is processed in the implementation since it needs to do branching on it.

What might be the use case for types besides `EntryValue` and `string`?

Reply:
To be honest I simply don't see the point of the `EntryValue` and of this API in general, which I find to be unnecessarily complicated. This is why I'm proposing an [alternative](https://github.com/open-telemetry/opentelemetry-specification/issues/140).

Reply:
> export type DistributedContext { [key: string]: EntryValue}

I really like the simplified type (Earlier I was thinking the same), but how would you achieve [validations/restrictions](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/distributedcontext-api.md#entrykey) on `EntryKey` and `EntryValue`.

Reply:
I'm not sure I understand why this wouldn't be possible with simplified types.

Reply:
Yeah, would it work to just do that validation at the point where the structure is received as input (say in the tracer, etc.)?

Reply:
I'm convinced with the simplified types, updated the PR. Please take a look again, will revisit after https://github.com/open-telemetry/opentelemetry-specification/issues/140